File: demo/test/report.test.js
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { Report } from '../src/report.js';

test('renders a report as plain text', () => {
  const report = new Report('Sales', [
    { label: 'Apples', value: 120 },
    { label: 'Oranges', value: 90 },
  ]);
  assert.equal(report.render(), 'Sales\n=====\nApples: 120\nOranges: 90');
});

File: demo-android/app/src/test/java/com/example/reports/ReportTest.kt
package com.example.reports

import org.junit.Assert.assertEquals
import org.junit.Test

class ReportTest {

    @Test
    fun rendersAReportAsPlainText() {
        val report = Report("Sales", listOf(Row("Apples", 120), Row("Oranges", 90)))
        assertEquals("Sales\n=====\nApples: 120\nOranges: 90", report.render())
    }

    @Test
    fun rendersAReportAsCsv() {
        val report = Report("Sales", listOf(Row("Apples", 120), Row("Oranges", 90)))
        assertEquals("label,value\nApples,120\nOranges,90", report.render("csv"))
    }
}
